theano/tensor/basic.py

```diff
@@ -5128,6 +5134,8 @@ class Diagonal(Op):
     """
 
     def __init__(self, offset=0, axis1=0, axis2=1):
+        if numpy_diagonal_return_view:
+            self.view_map = {0: [0]}
```

Comment:
view do not cause problem. I taught that at some point, (and wrongly changed DimShuffle for this). The problem is to have inplace in the user graph.